cms/djangoapps/contentstore/views/course.py

```diff
@@ -621,7 +621,7 @@ def get_courses_order_by(order_query, course_overviews):
         order_query (str): any string used to order Course Overviews.
         course_overviews (Course Overview objects): queryset to be ordered.
     """
-    if not order_query:
+    if not (order_query and course_overviews):
```

Comment:
What if `course_overviews` is actually a queryset, would this statement evaluate the queryset before the filtering?

Reply:
Evaluating `course_overviews` would incur additional costs. I think it’s better to annotate these functions so they don’t accept empty lists and only accept querysets.

Reply:
Do you think there's an in-between solution that wouldn't incur additional costs but would catch this type of inconsistency with lists vs querysets?

Reply:
To make it more centralized and minimize costs, we can add a condition for lists. For example:

```python
def get_courses_order_by(order_query, course_overviews):
    """
    Return course overviews based on a base queryset ordered by a query.

    Args:
        order_query (str): Any string used to order Course Overviews.
        course_overviews (QuerySet or list): Queryset to be ordered or a list of objects.

    Returns:
        QuerySet or list: Ordered queryset or the original list if ordering is not applicable.
    """
    if isinstance(course_overviews, list):
        log.warning("Expected a queryset but received a list object. Skipping ordering.")
        return course_overviews

    if not order_query:
        return course_overviews

    try:
        return course_overviews.order_by(order_query)
    except FieldError as e:
        log.exception(f"Error ordering courses by {order_query}: {e}")
        return course_overviews
```

However, your solution is much better because the function expects a queryset, not a list. I’ve reviewed the code, and this function is only used in `get_filtered_and_ordered_courses`, which is being used in `_accessible_courses_summary_iter` and `_accessible_courses_list_from_groups`. The case won’t occur in `_accessible_courses_summary_iter`, and your changes also validate this case in `_accessible_courses_list_from_groups`.